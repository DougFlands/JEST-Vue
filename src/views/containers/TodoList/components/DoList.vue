<template>
  <div class="dolist">
    <div>
      已完成:
      <span data-test="count">{{ list.length }}</span>
    </div>
    <ul class="content">
      <li v-for="(item, index) in list" :key="index" data-test="item">
        <div data-test="item-value" class="text">{{ item.value }}</div>
        <div>
          <span data-test="delete-button" @click="handleDelete(index)" class="btn">删除</span>
        </div>
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    list: {
      type: Array,
      default: () => []
    }
  },
  data() {
    return {
    };
  },

  created() {

  },

  methods: {
    handleDelete(index) {
      this.$emit('delete', index)
    },
  },
}

</script>

<style lang='scss' scoped>
.dolist {
  width: 600px;
  margin: 20px auto;
  .content {
    margin-top: 20px;
    li {
      display: flex;
      justify-content: space-between;
      margin-top: 10px;
    }
    .text {
      text-decoration: line-through;
    }
    .btn {
      cursor: pointer;
      margin-left: 20px;
      &:nth-child(1) {
        color: red;
      }
    }
  }
}
</style>