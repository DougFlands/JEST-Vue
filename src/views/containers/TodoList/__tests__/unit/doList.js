import { shallowMount } from '@vue/test-utils';
import DoList from '../../components/DoList.vue';
import { findTestWrapper } from '@/utils/testUtils'

it('DoList 样式改变则提示', () => {
  const wrapper = shallowMount(DoList)
  expect(wrapper).toMatchSnapshot()
})

it('DoList 参数为[], count 值为0，且列表无内容', () => {
  const wrapper = shallowMount(DoList, {
    propsData: {
      list: []
    }
  })
  const countElem = findTestWrapper(wrapper, 'count')
  const listItems = findTestWrapper(wrapper, 'item')
  expect(countElem.at(0).text()).toEqual('0')
  expect(listItems.length).toEqual(0)
})

it('DoList 参数为[1,2,3], count 值为3，列表有内容，且存在删除按钮', () => {
  const wrapper = shallowMount(DoList, {
    propsData: {
      list: [
        {
          status: 'div', value: 1,
        },
        {
          status: 'div', value: 2,
        },
        {
          status: 'div', value: 3,
        },
      ]
    }
  })
  const countElem = findTestWrapper(wrapper, 'count')
  const listItems = findTestWrapper(wrapper, 'item')
  const deleteButton = findTestWrapper(wrapper, 'delete-button')
  expect(countElem.at(0).text()).toEqual('3')
  expect(listItems.length).toEqual(3)
  expect(deleteButton.length).toEqual(3)
})

it('删除按钮点击时，触发删除事件', () => {
  const wrapper = shallowMount(DoList, {
    propsData: {
      list: [
        {
          status: 'div', value: 1,
        },
        {
          status: 'div', value: 2,
        },
        {
          status: 'div', value: 3,
        },
      ]
    }
  })
  const deleteButton = findTestWrapper(wrapper, 'delete-button').at(1)
  deleteButton.trigger('click')
  expect(wrapper.emitted().delete).toBeTruthy()
  // 传给 delete 的参数是 1
  expect(wrapper.emitted().delete[0][0]).toBe(1)
})