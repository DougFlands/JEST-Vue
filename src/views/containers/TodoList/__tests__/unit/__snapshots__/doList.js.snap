// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoList 样式改变则提示 1`] = `
<div class="dolist">
  <div>
    已完成:
    <span data-test="count">0</span></div>
  <ul class="content"></ul>
</div>
`;
